File: app/src/main/java/net/daverix/ajvm/jvm/IMulOperation.java
package net.daverix.ajvm.jvm;


import net.daverix.ajvm.Frame;
import net.daverix.ajvm.ByteCodeReader;

import java.io.IOException;

public class IMulOperation implements ByteCodeOperation {
    @Override
    public void execute(ByteCodeReader reader, int indexOfBytecode, Frame currentFrame) throws IOException {
        int iMul2 = (int) currentFrame.pop();
        int iMul1 = (int) currentFrame.pop();
        currentFrame.push(iMul1 * iMul2);
    }
}

File: app/src/main/java/net/daverix/ajvm/operation/IMulOperation.kt
package net.daverix.ajvm.operation


import net.daverix.ajvm.Frame
import net.daverix.ajvm.ByteCodeReader

import java.io.IOException

class IMulOperation : ByteCodeOperation {
    @Throws(IOException::class)
    override fun execute(reader: ByteCodeReader, indexOfBytecode: Int, currentFrame: Frame) {
        val iMul2 = currentFrame.pop() as Int
        val iMul1 = currentFrame.pop() as Int
        currentFrame.push(iMul1 * iMul2)
    }
}
